src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -715,15 +715,119 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
+                        frontend_logger.log("Error writing to socket", "error");
+                    }
+
+                    result_wr = write(connFd, "\r\n", 2);
+
+                    if (result_wr < 0) {
+                        frontend_logger.log("Error writing to socket", "error");
+                    }
+                    break;
+                }
+
+                int threadPriority = std::atoi(priority.c_str());
+
+                //All high priority threads will be set the same high priority level
+                if (threadPriority > Conts::DEFAULT_THREAD_PRIORITY) {
+                    threadPriority = Conts::HIGH_PRIORITY_DEFAULT_VALUE;
+
+                    if (highPriorityTaskCount > 0) {
+                        bool rejectJob = false;
+                        PerformanceUtil performanceUtil;
+                        performanceUtil.init();
+
+                        bool resourceSufficient = performanceUtil.isResourcesSufficient(graph_id,TRIANGLES,Conts::SLA_CATEGORY::LATENCY);
+                        int runningHPTaskCount = JasmineGraphFrontEnd::runningHighPriorityTasks();
+
+                        if (resourceSufficient && runningHPTaskCount == highPriorityTaskCount) {
+                            bool queueTimeAcceptable = JasmineGraphFrontEnd::isQueueTimeAcceptable(sqlite,perfSqlite,
+                                    graph_id,TRIANGLES,Conts::SLA_CATEGORY::LATENCY);
+
+                            if (!queueTimeAcceptable) {
+                                rejectJob = true;
+                            }
+
+                        } else {
+                            rejectJob = true;
+                        }
+
+                        if (rejectJob) {
+                            string error_message = "System has reached the maximum high priority tasks allowed "
+                                                   "at a given time. Please try again later.";
+                            result_wr = write(connFd, error_message.c_str(), error_message.length());
+
+                            if (result_wr < 0) {
+                                frontend_logger.log("Error writing to socket", "error");
+                            }
+
+                            result_wr = write(connFd, "\r\n", 2);
+
+                            if (result_wr < 0) {
+                                frontend_logger.log("Error writing to socket", "error");
+                            }
+                            break;
+                        }
+                    } else {
+                        highPriorityTaskCount++;
+                    }
+                }
+
                 auto begin = chrono::high_resolution_clock::now();
-                long triangleCount = JasmineGraphFrontEnd::countTriangles(graph_id, sqlite, perfSqlite, masterIP);
+                JobRequest jobDetails;
+                jobDetails.setJobId(std::to_string(uniqueId));
+                jobDetails.setJobType(TRIANGLES);
```

Comment:
Is this calibration specifically bound to triangle counting?

Reply:
Yes this is bound to triangle counting. Because the calibration decision will have to be taken in a flow specific way.

Reply:
Shall we allow the user to provide the types of the job from the frontend? That way users will be able to configure an run calibrations for different types of algorithms.

Reply:
After the offline discussion, it was understood that this is specif for triangle count.